saleor/graphql/product/tests/queries/test_product_variants_query_with_where.py

```diff
@@ -172,3 +173,103 @@ def test_product_variant_filter_by_sku(
     assert len(variants) == len(indexes)
     skus = {node["node"]["sku"] for node in variants}
     assert skus == {product_variant_list[index].sku for index in indexes}
+
+
+@pytest.mark.parametrize(
+    "stocks_to_create",
+    [[], [("1", 0), ("2", 1), ("4", 1)]],
+)
+def test_product_variant_where_in_stock(
+    api_client,
+    product_variant_list,
+    channel_USD,
+    stocks_to_create,
+    shipping_zones_with_warehouses,
+):
+    # given
+    warehouse = shipping_zones_with_warehouses[0].warehouses.first()
+    channel_USD.warehouses.add(warehouse)
+    Stock.objects.bulk_create(
+        [
+            Stock(
+                product_variant=ProductVariant.objects.filter(sku=sku).first(),
+                quantity=quantity,
+                warehouse=warehouse,
+            )
+            for sku, quantity in stocks_to_create
```

Comment:
Instead of fetching the variant from the database each time, consider using indexes to reference variants directly from the `product_variant_list`.